src/python/pants/backend/awslambda/python/rules.py

```diff
@@ -34,6 +39,20 @@ class PythonAwsLambdaFieldSet(PackageFieldSet):
 
     handler: PythonAwsLambdaHandlerField
     include_requirements: PythonAwsLambdaIncludeRequirements
+    include_sources: PythonAwsLambdaIncludeSources
+    runtime: PythonAwsLambdaRuntime
+    complete_platforms: PythonFaaSCompletePlatforms
+    output_path: OutputPathField
+    environment: EnvironmentField
+
+
+@dataclass(frozen=True)
+class PythonAwsLambdaLayerFieldSet(PackageFieldSet):
```

Comment:
This (and the rules below) end up rather non-DRY, but the repetition is generally uninteresting (just splat the fieldset field into the matching request field). Let me know if there's a good way to DRY it more.

Reply:
Shared base class for the FieldSet?

Reply:
I've done that, although the more interesting duplication is constructing the `BuildPythonFaaSRequest` and I don't know of a good way to DRY that.